10_construct-database/13_construct_county_race-share-geography.qmd

```diff
@@ -0,0 +1,302 @@
+---
+title: "Combine Race-Share Geography County Files"
+author: "Aaron R. Williams & JP Walsh"
+date: today
+format:
+  html:
+    embed-resources: true
+    toc: true
+    toc_float: true
+execute:
+  message: false
+  warning: false
+editor_options: 
+  chunk_output_type: console
+---
+
+## Setup
+
+```{r quarto-setup, include = FALSE}
+options(knitr.kable.NA = "")
+
+```
+
+```{r load-packages}
+library(tidyverse)
+library(tidylog)
+library(here)
+library(assertr)
+library(gt)
+
+```
+
+This function is used to ensure that the lower bound of a 95% confidence interval is always below the estimate and the upper bound of the a 95% confidence interval is always above the estimate. 
+
+```{r}
+#' Test the bounds of a confidence interval relative to the estimate
+#'
+#' @param data The data frame of interest
+#' @param estimate The unquoted name of the estimate variable
+#' @param lb The unquoted name of the lower bound variable
+#' @param ub The unquoted name of the upper bound variable
+#'
+test_bounds <- function(data, estimate, lb, ub) {
+  
+  subset <- bind_rows(
+    data |>
+      filter({{ ub }} < {{ lb }}),
+    data |>
+      filter({{ estimate }} > {{ ub }}),
+    data |>
+      filter({{ estimate  }} < {{ lb }}),
+  )
+  
+  stopifnot(nrow(subset) == 0)
+  
+}
+
+
+#' Helper function to silence output from testing code
+#'
+#' @param data A data frame
+#'
+quiet <- function(data) {
+  
+  quiet <- data
+  
+}
+
+```
+
+## Construct Database
+
+### Temporary Fixes
+
+This section implements temporary fixes and saves the resulting data in a temp folder. The saved data are then included in the output data.
+
+```{r}
+# combine multiple years
+bind_rows(
+  read_csv(here("01_financial-well-being/share_debt_2018_long.csv")) |>
+    select(-county_name),
+  read_csv(here("01_financial-well-being/county-debt-coll-shares-2022.csv"))
+  ) |> 
+  mutate(share_debt_coll_quality = ifelse(is.na(share_debt_coll), NA_character_, share_debt_coll_quality))|>
+  write_csv(here("data", "temp", "county_debt_longitudinal_race-ethnicity.csv"))
+
+# Correct subgroup category
+
+read_csv(here("data", "temp", "county_debt_longitudinal_race-ethnicity.csv")) |>
+  mutate(subgroup = case_when(subgroup == "Majority White" ~ "Majority White, Non-Hispanic",
+                              subgroup == "No Majority Race/Ethnicity" ~ "Mixed Race and Ethnicity", 
+                              subgroup == "Majority Non-White" ~ "Majority Households of Color",
+                              TRUE ~ subgroup),
+         subgroup_type = ifelse(subgroup == "All", "all", subgroup_type)) |>
+  write_csv(
+    here("data", "temp", "county_debt_longitudinal_subgroup_race_share.csv")
+  )
```

Comment:
The debt metric is as the person level. The race/ethnicity share is at the person level. I don't think Households makes sense here.